public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx

```diff
@@ -0,0 +1,126 @@
+import { css, cx } from '@emotion/css';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { CustomScrollbar, Icon, useTheme2 } from '@grafana/ui';
+
+export interface Props {
+  children: React.ReactNode;
+}
+
+export const NavBarScrollContainer = ({ children }: Props) => {
+  const [showScrollTopIndicator, setShowTopScrollIndicator] = useState(false);
+  const [showScrollBottomIndicator, setShowBottomScrollIndicator] = useState(false);
+  const scrollTopMarker = useRef<HTMLDivElement>(null);
+  const scrollBottomMarker = useRef<HTMLDivElement>(null);
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+
+  // Here we observe the top and bottom markers to determine if we should show the scroll indicators
+  useEffect(() => {
+    const intersectionObserver = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.target === scrollTopMarker.current) {
+          setShowTopScrollIndicator(!entry.isIntersecting);
+        } else if (entry.target === scrollBottomMarker.current) {
+          setShowBottomScrollIndicator(!entry.isIntersecting);
+        }
+      });
+    });
+    [scrollTopMarker, scrollBottomMarker].forEach((ref) => {
+      if (ref.current) {
+        intersectionObserver.observe(ref.current);
+      }
+    });
+    return () => intersectionObserver.disconnect();
+  }, []);
+
+  return (
+    <CustomScrollbar className={styles.scrollContainer} hideVerticalTrack hideHorizontalTrack>
+      <div
+        className={cx(styles.scrollTopIndicator, {
+          [styles.scrollIndicatorVisible]: showScrollTopIndicator,
+        })}
+      >
+        <Icon className={styles.scrollTopIcon} name="angle-up" />
+      </div>
+      <div className={styles.scrollContent}>
+        <div className={styles.scrollTopMarker} ref={scrollTopMarker}></div>
+        {children}
+        <div className={styles.scrollBottomMarker} ref={scrollBottomMarker}></div>
+      </div>
+      <div
+        className={cx(styles.scrollBottomIndicator, {
+          [styles.scrollIndicatorVisible]: showScrollBottomIndicator,
+        })}
+      >
+        <Icon className={styles.scrollBottomIcon} name="angle-down" />
+      </div>
+    </CustomScrollbar>
+  );
+};
+
+NavBarScrollContainer.displayName = 'NavBarScrollContainer';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  'scrollTopMarker, scrollBottomMarker': css({
+    height: theme.spacing(1),
+    left: 0,
+    position: 'absolute',
+    pointerEvents: 'none',
+    right: 0,
+  }),
+  scrollTopMarker: css({
+    top: 0,
+  }),
+  scrollBottomMarker: css({
+    bottom: 0,
+  }),
+  scrollContent: css({
+    position: 'relative',
+  }),
+  scrollContainer: css`
+    .scrollbar-view {
+      position: static !important;
```

Comment:
Could you add a comment here for why this `!important` is needed? Always wary of these 😅

Reply:
yeah i don't like it either. if we can find a better way i'd love to change it

Reply:
Yeah that's fair, thanks for adding the comment!